pkg/server/application_api/application_api_test: TestAdminAPIDataDistribution failed

pkg/server/application_api/application_api_test.TestAdminAPIDataDistribution [failed](https://tanzanite.cluster.engflow.com/invocations/default/e36cf312-f9d0-4b48-b794-2a726262da81?testReportRun=2&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyL2FwcGxpY2F0aW9uX2FwaTphcHBsaWNhdGlvbl9hcGlfdGVzdA==) on master @ [e215cf435851cb9490fe4494fe73d02041c4d2c0](https://github.com/cockroachdb/cockroach/commits/e215cf435851cb9490fe4494fe73d02041c4d2c0):


```
=== RUN   TestAdminAPIDataDistribution
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestAdminAPIDataDistribution131073793
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:93: cluster virtualization disabled due to issue: #109501 (expected label: C-bug)
    storage_inspection_test.go:314: Get "https://127.0.0.1:34167/_admin/v1/data_distribution": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestAdminAPIDataDistribution131073793
--- FAIL: TestAdminAPIDataDistribution (77.91s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121889 pkg/server/application_api/application_api_test: TestAdminAPIDataDistribution failed [C-test-failure O-robot P-2 branch-release-24.1]
</p>
</details>
/cc @cockroachdb/unowned
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdminAPIDataDistribution.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-38113

pkg/server/application_api/application_api_test.TestAdminAPIDataDistribution [failed](https://tanzanite.cluster.engflow.com/invocations/default/c3f7467e-a0e1-417c-947c-affcf23291c9?testReportRun=3&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyL2FwcGxpY2F0aW9uX2FwaTphcHBsaWNhdGlvbl9hcGlfdGVzdA==) on master @ [af3173a899e16af30d2ca9489d8c2c926c410121](https://github.com/cockroachdb/cockroach/commits/af3173a899e16af30d2ca9489d8c2c926c410121):


```
=== RUN   TestAdminAPIDataDistribution
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestAdminAPIDataDistribution823364192
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:93: cluster virtualization disabled due to issue: #109501 (expected label: C-bug)
    storage_inspection_test.go:281: Get "https://127.0.0.1:36607/_admin/v1/data_distribution": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestAdminAPIDataDistribution823364192
--- FAIL: TestAdminAPIDataDistribution (52.49s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121889 pkg/server/application_api/application_api_test: TestAdminAPIDataDistribution failed [C-test-failure O-robot P-2 branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdminAPIDataDistribution.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/server/application_api/application_api_test.TestAdminAPIDataDistribution [failed](https://tanzanite.cluster.engflow.com/invocations/default/d6a24199-bbf0-4ef7-b2b8-a78af1e362dd?testReportRun=2&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyL2FwcGxpY2F0aW9uX2FwaTphcHBsaWNhdGlvbl9hcGlfdGVzdA==) on master @ [379d332c9a4ce49cda4a3565a852e7ddb850ffb5](https://github.com/cockroachdb/cockroach/commits/379d332c9a4ce49cda4a3565a852e7ddb850ffb5):


```
=== RUN   TestAdminAPIDataDistribution
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestAdminAPIDataDistribution287537023
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:93: cluster virtualization disabled due to issue: #109501 (expected label: C-bug)
    storage_inspection_test.go:281: Get "https://127.0.0.1:40599/_admin/v1/data_distribution": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestAdminAPIDataDistribution287537023
--- FAIL: TestAdminAPIDataDistribution (57.26s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121889 pkg/server/application_api/application_api_test: TestAdminAPIDataDistribution failed [C-test-failure O-robot P-2 branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdminAPIDataDistribution.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/server/application_api/application_api_test.TestAdminAPIDataDistribution [failed](https://tanzanite.cluster.engflow.com/invocations/default/05c32f57-330c-4aaf-a1bd-8d78047ceb97?testReportRun=2&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyL2FwcGxpY2F0aW9uX2FwaTphcHBsaWNhdGlvbl9hcGlfdGVzdA==) on master @ [c6b012b6d77c8fda71bd9c4dd24570895f32cbbe](https://github.com/cockroachdb/cockroach/commits/c6b012b6d77c8fda71bd9c4dd24570895f32cbbe):


```
=== RUN   TestAdminAPIDataDistribution
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestAdminAPIDataDistribution1287949989
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:93: cluster virtualization disabled due to issue: #109501 (expected label: C-bug)
    storage_inspection_test.go:281: Get "https://127.0.0.1:41425/_admin/v1/data_distribution": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestAdminAPIDataDistribution1287949989
--- FAIL: TestAdminAPIDataDistribution (59.72s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121889 pkg/server/application_api/application_api_test: TestAdminAPIDataDistribution failed [C-test-failure O-robot P-2 branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdminAPIDataDistribution.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/server/application_api/application_api_test.TestAdminAPIDataDistribution [failed](https://tanzanite.cluster.engflow.com/invocations/default/de8ac542-e0a8-4034-9df2-f2562e7787e9?testReportRun=1&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyL2FwcGxpY2F0aW9uX2FwaTphcHBsaWNhdGlvbl9hcGlfdGVzdA==) on master @ [e4d744226f47e37b9179632711067277b2b0ab80](https://github.com/cockroachdb/cockroach/commits/e4d744226f47e37b9179632711067277b2b0ab80):


```
=== RUN   TestAdminAPIDataDistribution
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestAdminAPIDataDistribution2390734564
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:93: cluster virtualization disabled due to issue: #109501 (expected label: C-bug)
    storage_inspection_test.go:303: Get "https://127.0.0.1:33481/_admin/v1/data_distribution": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestAdminAPIDataDistribution2390734564
--- FAIL: TestAdminAPIDataDistribution (62.75s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121889 pkg/server/application_api/application_api_test: TestAdminAPIDataDistribution failed [C-test-failure O-robot P-2 branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdminAPIDataDistribution.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/server/application_api/application_api_test.TestAdminAPIDataDistribution [failed](https://tanzanite.cluster.engflow.com/invocations/default/a9334f31-feea-42c0-bd70-0e0bb65265ac?testReportRun=1&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyL2FwcGxpY2F0aW9uX2FwaTphcHBsaWNhdGlvbl9hcGlfdGVzdA==) on master @ [74d5e3e8021e53d27d823ef7f8694d59df8c3414](https://github.com/cockroachdb/cockroach/commits/74d5e3e8021e53d27d823ef7f8694d59df8c3414):


```
=== RUN   TestAdminAPIDataDistribution
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestAdminAPIDataDistribution1091907618
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:93: cluster virtualization disabled due to issue: #109501 (expected label: C-bug)
    storage_inspection_test.go:281: Get "https://127.0.0.1:45733/_admin/v1/data_distribution": net/http: request canceled (Client.Timeout exceeded while awaiting headers)
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestAdminAPIDataDistribution1091907618
--- FAIL: TestAdminAPIDataDistribution (77.78s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #121889 pkg/server/application_api/application_api_test: TestAdminAPIDataDistribution failed [C-test-failure O-robot P-2 branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAdminAPIDataDistribution.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/server/testserver_http.go
+++ b/pkg/server/testserver_http.go
@@ -18,6 +18,7 @@ import (
 	"net/http/cookiejar"
 	"net/url"
 	"sync"
+	"time"
 
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/security/username"
@@ -26,6 +27,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/server/serverpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/util"
 	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
 )
 
@@ -88,6 +90,9 @@ func (ts *httpTestServer) GetUnauthenticatedHTTPClient() (http.Client, error) {
 		RoundTripper: client.Transport,
 		tenantName:   ts.t.tenantName,
 	}
+	if util.RaceEnabled {
+		client.Timeout = 30 * time.Second
+	}
 	return client, nil
 }
 
@@ -96,6 +101,9 @@ func (ts *httpTestServer) GetAdminHTTPClient() (http.Client, error) {
 	httpClient, _, err := ts.GetAuthenticatedHTTPClientAndCookie(
 		apiconstants.TestingUserName(), true, serverutils.SingleTenantSession,
 	)
+	if util.RaceEnabled {
+		httpClient.Timeout = 30 * time.Second
+	}
 	return httpClient, err
 }
 
@@ -108,6 +116,9 @@ func (ts *httpTestServer) GetAuthenticatedHTTPClient(
 		authUser = apiconstants.TestingUserNameNoAdmin()
 	}
 	httpClient, _, err := ts.GetAuthenticatedHTTPClientAndCookie(authUser, isAdmin, session)
+	if util.RaceEnabled {
+		httpClient.Timeout = 30 * time.Second
+	}
 	return httpClient, err
 }
 

